doc/internals/how-to-add-new-backend.rst

```diff
@@ -267,42 +268,57 @@ interface only the boolean keywords related to the supported decoders.
 How to register a backend
 +++++++++++++++++++++++++
 
-Define a new entrypoint in your ``setup.py`` (or ``setup.cfg``) with:
+Define a new entrypoint in your ``pyproject.toml`` (or ``setup.cfg/setup.py`` for older
+configurations), with:
 
 - group: ``xarray.backends``
 - name: the name to be passed to :py:meth:`~xarray.open_dataset`  as ``engine``
 - object reference: the reference of the class that you have implemented.
 
-You can declare the entrypoint in ``setup.py`` using the following syntax:
+You can declare the entrypoint in your project configuration like so:
 
-.. code-block::
+.. tab:: pyproject.toml
 
-    setuptools.setup(
-        entry_points={
-            "xarray.backends": ["my_engine=my_package.my_module:MyBackendEntryClass"],
-        },
-    )
+   .. code:: toml
+
+      [project.entry-points."xarray-backends"]
+      my_engine = "my_package.my_module:MyBackendEntryClass"
```

Comment:
Actually maybe this should be 

```suggestion
      my_engine = "my_package.my_module:MyBackendEntrypoint"
```

to match [the example](https://docs.xarray.dev/en/latest/internals/how-to-add-new-backend.html#backendentrypoint-subclassing)?

Reply:
Yes lets do that. (I can't apply that change for some reason)_

Reply:
I think that's also caused by access restrictions to the repositories in organizations (not sure though).

Reply:
Hmm, thought I had allowed maintainers to modify the branch -- no worries, I've made the change! :)